apps/api/src/app/events/usecases/filter-steps/filter-steps-backoff.usecase.ts

```diff
@@ -0,0 +1,77 @@
+import { Injectable } from '@nestjs/common';
+import { JobStatusEnum, JobRepository, NotificationStepEntity } from '@novu/dal';
+import { ChannelTypeEnum } from '@novu/shared';
+import { FilterStepsCommand } from './filter-steps.command';
+import * as moment from 'moment';
+import { FilterSteps } from './filter-steps.usecase';
+
+@Injectable()
+export class FilterStepsBackoff {
+  constructor(private jobRepository: JobRepository) {}
+
+  public async execute(command: FilterStepsCommand): Promise<NotificationStepEntity[]> {
+    const steps = [FilterSteps.createTriggerStep(command)];
+    for (const step of command.steps) {
+      if (step.template.type === ChannelTypeEnum.DIGEST) {
+        const trigger = await this.getTrigger(command, step);
+        if (!trigger) {
+          continue;
+        }
+
+        const digests = await this.getDigests(command, step);
+
+        if (digests.length > 0) {
+          return steps;
```

Comment:
what do you have in mind?

Reply:
Nothing particular, just to make the if statement contents extracted to a function so it would be easier to read. Really up to you, it's fine right now, but I missed the end of the if statement in the first time reading it and had to find the opening of it to understand it.